lib/mix/lib/mix/tasks/local.hex.ex

```diff
@@ -33,7 +33,7 @@ defmodule Mix.Tasks.Local.Hex do
 
   ## Mirrors
 
-  If you want to change the [default mirror](https://repo.hex.pm)
+  If you want to change the [default mirror](https://builds.hex.pm)
   used for fetching Hex, set the `HEX_MIRROR` environment variable.
```

Comment:
Should we change the name of this environment variable? Hex uses it for the repository but now it will be used for builds in Elixir.

Reply:
I mean, if you remove the redirect at some point then you need to decouple, no?

Reply:
I mean that Hex uses `HEX_MIRROR` to fetch packages and registry information, the default is repo.hex.pm.

With this change Elixir uses `HEX_MIRROR` to fetch pre-compiled Hex and Rebar and the default will be builds.hex.pm.

So we have the same env var used for different reasons and with different default values in Elixir and Hex.

Reply:
Yeah, we likely want a HEX_BUILDS and fallback to HEX_MIRROR.